src/Autofac.Integration.Mef/RegistrationExtensions.cs

```diff
@@ -423,10 +427,136 @@ private static void ProcessExportDefinition(ContainerBuilder builder, Func<Expor
+                    {
+                        // This is either a getter or a setter.
+                        resultType = m.ReturnType;
+                        if (resultType == typeof(void))
+                        {
+                            resultType = m.GetParameters()[0].ParameterType;
+                        }
+
+                        break;
+                    }
+                    else if (v is FieldInfo f)
+                    {
+                        resultType = f.FieldType;
+                        break;
+                    }
+                }
+            }
+
+            if (resultType != null && resultType != typeof(void))
+            {
+                // Have to handle 2 cases
+                // Single cardinality = Lazy<T, TMetadata>
+                // Multiple cardinality = IEnumerable<Lazy<T, TMetadata>>
+                bool isLazy;
+                if (
+                    (isLazy = resultType.IsGenericTypeDefinedBy(typeof(Lazy<,>)))
+                        ||
+                    (
+                    resultType.IsGenericTypeDefinedBy(typeof(IEnumerable<>))
+                        &&
+                    resultType.GetGenericArguments()[0].IsGenericTypeDefinedBy(typeof(Lazy<,>))))
+                {
+                    lazyType = isLazy ? resultType : resultType.GetGenericArguments()[0];
+                    var objectType = lazyType.GetGenericArguments()[0];
+
+                    // Resolve as Lazy<T, IDictionary<string, object>> so we can leverage the Metadata value to populate
+                    // the metadata on the Exports built later.
+                    // If we do not change the type here, we end up loosing the name/value pairs so the resulting
+                    // Metadata will not be populated
+                    lazyType = lazyType
+                        .GetGenericTypeDefinition()
+                        .MakeGenericType(objectType, typeof(IDictionary<string, object>));
+
+                    if (!isLazy)
+                    {
+                        resultType = typeof(IEnumerable<>).MakeGenericType(lazyType);
+                    }
+                    else
+                    {
+                        resultType = lazyType;
+                    }
+
+                    return true;
+                }
+            }
+
+            resultType = null;
+            lazyType = null;
+            return false;
+        }
+
         private static IEnumerable<Export> ResolveExports(this IComponentContext context, ContractBasedImportDefinition definition)
         {
             var contractService = new ContractBasedService(definition.ContractName, definition.RequiredTypeIdentity);
 
+            if (definition.TryGetLazyType(out var resultType, out var lazyType))
+            {
+                if (context.TryResolve(resultType, out var resolved))
+                {
+                    var valueProperty = lazyType.GetProperty(nameof(Lazy<int, int>.Value));
```

Comment:
It was just a habit I started to reduce the likelihood of things being missed due to being accessed via Reflection/Magic strings since the nameof() keyword is a reference that can be investigated via "Find all references"

Ideally it would have been nameof(Lazy<,>.Value) but that doesn't compile unfortunately.

Updated to be a little more clear.